framework/src/tinyalsa/tinyalsa.c

```diff
@@ -295,6 +290,7 @@ const char *pcm_get_error(const struct pcm *pcm)
 int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
 {
```

Comment:
I think we should make pcm_set_config as static function, i.e. remove it as a public API. 
Because a call to pcm_set_config after pcm_open has been called would lead to errors because buffers are allocated in pcm_open according to the config passed in pcm_open. Various fields of pcm struct will be changed without changing buffer allocation .